tests/bluetooth/bsim_bt/bsim_test_audio/src/mics_test.c

```diff
@@ -0,0 +1,494 @@
+static volatile uint8_t g_aics_input_mute;
+static volatile uint8_t g_aics_mode;
+static volatile uint8_t g_aics_input_type;
+static volatile uint8_t g_aics_units;
+static volatile uint8_t g_aics_gain_max;
+static volatile uint8_t g_aics_gain_min;
+static volatile bool g_aics_active = true;
+static char g_aics_desc[AICS_DESC_SIZE];
+static volatile bool g_cb;
+static struct bt_conn *g_conn;
+static bool g_is_connected;
+
+static void mics_mute_cb(struct bt_conn *conn, int err, uint8_t mute)
+{
+	if (err != 0) {
+		FAIL("MICS mute cb err (%d)", err);
+		return;
+	}
+
+	g_mute = mute;
+
+	if (conn == NULL) {
+		g_cb = true;
+	}
+}
+
+static void aics_state_cb(struct bt_conn *conn, struct bt_aics *inst, int err,
+			  int8_t gain, uint8_t mute, uint8_t mode)
+{
+	if (err != 0) {
+		FAIL("AICS state cb err (%d)", err);
+		return;
+	}
+
+	g_aics_gain = gain;
+	g_aics_input_mute = mute;
+	g_aics_mode = mode;
+
+	if (conn == NULL) {
+		g_cb = true;
+	}
+}
+
+static void aics_gain_setting_cb(struct bt_conn *conn, struct bt_aics *inst,
+				 int err, uint8_t units, int8_t minimum,
+				 int8_t maximum)
+{
+	if (err != 0) {
+		FAIL("AICS gain setting cb err (%d)", err);
+		return;
+	}
+
+	g_aics_units = units;
+	g_aics_gain_min = minimum;
+	g_aics_gain_max = maximum;
+
+	if (conn == NULL) {
+		g_cb = true;
+	}
+}
+
+static void aics_input_type_cb(struct bt_conn *conn, struct bt_aics *inst,
+			       int err, uint8_t input_type)
+{
+	if (err != 0) {
+		FAIL("AICS input type cb err (%d)", err);
+		return;
+	}
+
+	g_aics_input_type = input_type;
+
+	if (conn == NULL) {
+		g_cb = true;
+	}
+}
+
+static void aics_status_cb(struct bt_conn *conn, struct bt_aics *inst, int err,
+			   bool active)
+{
+	if (err != 0) {
+		FAIL("AICS status cb err (%d)", err);
+		return;
+	}
+
+	g_aics_active = active;
+
+	if (conn == NULL) {
+		g_cb = true;
+	}
+}
+
+static void aics_description_cb(struct bt_conn *conn, struct bt_aics *inst,
+				int err, char *description)
+{
+	if (err != 0) {
+		FAIL("AICS description cb err (%d)", err);
+		return;
+	}
+
+
+	strncpy(g_aics_desc, description, sizeof(g_aics_desc) - 1);
+	g_aics_desc[sizeof(g_aics_desc) - 1] = '\0';
+
+	if (conn == NULL) {
+		g_cb = true;
+	}
+}
+
+static struct bt_mics_cb mics_cb = {
+	.mute = mics_mute_cb,
+};
+
+static struct bt_aics_cb aics_cb = {
+	.state = aics_state_cb,
+	.gain_setting = aics_gain_setting_cb,
+	.type = aics_input_type_cb,
+	.status = aics_status_cb,
+	.description = aics_description_cb
+};
+
+static void connected(struct bt_conn *conn, uint8_t err)
+{
+	char addr[BT_ADDR_LE_STR_LEN];
+
+	bt_addr_le_to_str(bt_conn_get_dst(conn), addr, sizeof(addr));
+
+	if (err != 0) {
+		FAIL("Failed to connect to %s (%u)\n", addr, err);
+		return;
+	}
+
+	printk("Connected to %s\n", addr);
+	g_conn = conn;
+	g_is_connected = true;
+}
+
+static struct bt_conn_cb conn_callbacks = {
+	.connected = connected,
+	.disconnected = disconnected,
+};
+
+static int test_aics_standalone(void)
```

Comment:
Probably a copy and paste, but it is slightly misleading to call this "test_aics_standalone" when it is in fact testing aics not standalone but as part of mics.

Reply:
yes